### PR TITLE
Feat/xml upload page

### PR DESCRIPTION
**Pull request by lustigy96**

### CODE REVIEW

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

File: src/pages/XmlUploadSite/components/ProgressBox/ProgressBoxes.jsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+              aria-label='cancel upload'
+              component='span'
+            >
+              <CloseIcon className={classes.closeIcon} />
+            </IconButton>
+          )}
+        </Box>
+        <Typography className={classes.typography}>
+          {`${Math.round(
+            file.size / 1000,
```
Comment (carmeltsadok) [inline src/pages/XmlUploadSite/components/ProgressBox/ProgressBoxes.jsx:40]: Calculate from const

File: src/pages/XmlUploadSite/components/ProgressBox/ProgressBoxes.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import useStyles from './ProgressBoxes.styles';
+import xmlImg from '../../images/xmlFile.png';
+
+const ProgressBoxes = ({ files, setFiles }) => {
+  const classes = useStyles();
+  const deleteFromList = (fileToDelete) => {
+    setFiles([...files.filter((f) => f.path !== fileToDelete.path)]);
+  };
+
+  const progressBox = (file) => (
```
Comment (carmeltsadok) [inline src/pages/XmlUploadSite/components/ProgressBox/ProgressBoxes.jsx:17]: Move to a new component

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

File: src/pages/XmlUploadSite/components/ProgressBox/ProgressBoxes.styles.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    fontSize: '0.7rem',
+    fontFamily: theme.typography.fontFamily,
+    color: 'gray',
+    marginBottom: '0.4rem',
+    textAlign: 'left',
+  },
+  progressBar: {
+    backgroundColor: 'rgb(234,243,246)',
+  },
+  progressBarColor: {
```
[the remark this answers is not in the document]
Comment (lustigy96) [reply src/pages/XmlUploadSite/components/ProgressBox/ProgressBoxes.styles.js:57]: why ? (talk)

File: src/pages/XmlUploadSite/components/DropZone/DropZone.jsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  };
+
+  return (
+    <Dropzone
+      onDrop={handleOnDrop}
+      accept={configPage.dropzone.acceptedFileTypes}
+      maxSize={configPage.dropzone.maxSize}
+    >
+      {({ getRootProps, getInputProps }) => (
+        /* eslint-disable react/jsx-props-no-spreading */
```
[the remark this answers is not in the document]
Comment (lustigy96) [reply src/pages/XmlUploadSite/components/DropZone/DropZone.jsx:71]: let's work through this together

File: src/pages/XmlUploadSite/components/ProgressBox/ProgressBoxes.jsx (hunk 2 above)
Comment (lustigy96) [reply src/pages/XmlUploadSite/components/ProgressBox/ProgressBoxes.jsx:17]: Without it, the progressBoxes is really tiny, so I dont think that it is needed

File: src/pages/XmlUploadSite/components/ProgressBox/ProgressBoxes.jsx (hunk 1 above)
Comment (lustigy96) [reply src/pages/XmlUploadSite/components/ProgressBox/ProgressBoxes.jsx:40]: why

File: src/theme.js [hunk 5]
```diff
@@ -11,11 +11,12 @@ const theme = createMuiTheme({
...
+      main: 'rgb(251,179,53)',
     },
     section: {
-      main: 'rgb(71,159,251)',
-      primary: 'rgb(238,246,255)',
+      main: 'rgb(247,251,252)',
+      primary: 'rgb(16,171,196)',
+      secondary: 'rgb(251,179,53)',
     },
   },
```
[the remark this answers is not in the document]
Comment (lustigy96) [reply src/theme.js:21]: compare

File: mockserver/index.js [hunk 6]
```diff
@@ -34,5 +35,6 @@ app.get('/config', (req, res) => {
 });
 
 app.use('/api/event', eventRouter);
+app.use('/api/upload-xml-file', xmlRouter);
```
Comment (carmeltsadok) [inline mockserver/index.js:38]:
```suggestion
app.use('/api/xml-upload', xmlRouter);
```

Comment (carmeltsadok) [reply mockserver/index.js:38]: According to the api

File: src/pages/XmlUploadSite/components/ProgressBox/ProgressBoxes.jsx (hunk 2 above)
Comment (carmeltsadok) [reply src/pages/XmlUploadSite/components/ProgressBox/ProgressBoxes.jsx:17]: It's more readable when it is separated into tiny files

File: src/pages/XmlUploadSite/components/ProgressBox/ProgressBoxes.jsx (hunk 1 above)
Comment (carmeltsadok) [reply src/pages/XmlUploadSite/components/ProgressBox/ProgressBoxes.jsx:40]: So we will get the file limit from the config server

Comment (lustigy96) [reply src/pages/XmlUploadSite/components/ProgressBox/ProgressBoxes.jsx:40]: We may get the file limit from a config server, but it is irrelevant to this calculation. This calculation is about the uploaded file. The size limit is defined somewhere else (in const :) and we can take it from a config server

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

**Review by carmeltsadok — APPROVED** [review]

**Pull request merged**